File: src/main/java/net/averkhoglyad/chess/manager/core/service/LichessIntegrationService.java
package net.averkhoglyad.chess.manager.core.service;

import net.averkhoglyad.chess.manager.core.data.Paging;
import net.averkhoglyad.chess.manager.core.sdk.data.Game;
import net.averkhoglyad.chess.manager.core.sdk.data.PageResult;
import net.averkhoglyad.chess.manager.core.sdk.data.User;

import java.util.Optional;

public interface LichessIntegrationService {

    Optional<User> getUser(String username);

    Optional<PageResult<Game>> getUserGames(String username, Paging paging);

    Optional<Game> getGame(String gameId);

    String loadGamePgn(String gameId);

}

File: src/main/kotlin/net/averkhoglyad/chess/manager/core/service/LichessIntegrationService.kt
package net.averkhoglyad.chess.manager.core.service

import net.averkhoglyad.chess.manager.core.data.Paging
import net.averkhoglyad.chess.manager.core.sdk.lichess.data.Game
import net.averkhoglyad.chess.manager.core.sdk.lichess.data.PageResult
import net.averkhoglyad.chess.manager.core.sdk.lichess.data.User

interface LichessIntegrationService {

    fun getUser(username: String): User?

    fun getUserGames(username: String, paging: Paging): PageResult<Game>?

    fun getGame(gameId: String): Game?

    fun loadGamePgn(gameId: String): String

}
